platform/linux-generic/odp_ipsec.c

```diff
@@ -1498,6 +1604,16 @@ static void ipsec_out_ah_post(ipsec_state_t *state, odp_packet_t pkt)
 		ipv6hdr->ver_tc_flow = state->ah_ipv6.ver_tc_flow;
 		ipv6hdr->hop_limit = state->ah_ipv6.hop_limit;
 	}
+
+	/* As ICV is already computed, truncate the ESN higher 32 bits. */
+	if (ipsec_sa->seqh_len) {
+		if (odp_packet_trunc_tail(&pkt, ipsec_sa->seqh_len, NULL, NULL) < 0) {
```

Comment:
This may change the packet handle but the changed handle is not propagated back to the caller who would continue operating with a stale handle.

Reply:
This API is used some other places in the same way. It seems that the new and old handles are the same.